### PR TITLE
egs2/iwslt22_dialect

### PR DESCRIPTION
**Pull request by brianyan918**
This PR is for the IWSLT22 Dialectal ST Track: https://iwslt.org/2022/dialect

The contents will include

- [x] ST data prep of Tunisian-Arabic to English triplet ST data
- [x] ST Baseline model recipes
- [x] ASR data prep and baseline recipes
- [x] Hugginface upload for st template

### CODE REVIEW

**Comment by ftshijt** [comment]
Looks cool! Just to remind:
1. please add an entry at egs2/README.md for the recipe
2. recommend to have an asr recipe as well, pre-training seems to be very important in egs2 (for both transformer and conformer)

**Comment by brianyan918** [comment]
Thanks! Yes I’m working on the asr1 portion now. Will update this.

**Comment by ftshijt** [comment]
Could you pull the latest master in espnet so as to fix the broken CI?

**Comment by sw005320** [comment]
Can you apply `black`?
https://github.com/espnet/espnet/runs/5072459563?check_suite_focus=true#step:8:10

**Comment by brianyan918** [comment]
Sorry, I fail some CI's but I do not understand them.. @ftshijt can you help me?

**Comment by sw005320** [comment]
We have a CI issue in #4043, which is fixed now.
So, could you merge the master?

If it is ready for merge, can you remove WIP in the subject?

File: egs2/iwslt22_dialect/asr1/RESULTS.md [hunk 1]
```diff
@@ -0,0 +1,47 @@
+<!-- Generated by scripts/utils/show_asr_result.sh -->
+# RESULTS
+## Environments
+- date: `Wed Feb  2 05:32:30 EST 2022`
+- python version: `3.8.12 (default, Oct 12 2021, 13:49:34)  [GCC 7.5.0]`
+- espnet version: `espnet 0.10.6a1`
+- pytorch version: `pytorch 1.8.1`
+- Git hash: `99581e0f5af3ad68851d556645e7292771436df9`
+  - Commit date: `Sat Jan 29 11:32:38 2022 -0500`
+
```
Comment (sw005320) [inline egs2/iwslt22_dialect/asr1/RESULTS.md:10]: Please upload a model to HF hub

File: egs2/iwslt22_dialect/asr1/conf/tuning/train_asr_conformer_ctc0.3_lr2e-3_warmup15k_newspecaug.yaml [hunk 2]
```diff
@@ -0,0 +1,73 @@
+batch_type: numel
```
Comment (sw005320) [inline egs2/iwslt22_dialect/asr1/conf/tuning/train_asr_conformer_ctc0.3_lr2e-3_warmup15k_newspecaug.yaml:1]: Is there any reason that you keep two configs?
If so, can you add the results of both configs to README?

File: egs2/iwslt22_dialect/st1/RESULT.md [hunk 3]
```diff
@@ -0,0 +1,23 @@
+# RESULTS
+## Environments
+- date: `Wed Feb  2 05:32:30 EST 2022`
+- python version: `3.8.12 (default, Oct 12 2021, 13:49:34)  [GCC 7.5.0]`
+- espnet version: `espnet 0.10.6a1`
+- pytorch version: `pytorch 1.8.1`
+- Git hash: `99581e0f5af3ad68851d556645e7292771436df9`
+  - Commit date: `Sat Jan 29 11:32:38 2022 -0500`
+
```
Comment (sw005320) [inline egs2/iwslt22_dialect/st1/RESULT.md:9]: ditto (HF hub)

File: egs2/iwslt22_dialect/st1/run.sh [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+train_dev=dev
+test_set=test1
+
+st_config=conf/train_st_conformer.yaml
+inference_config=conf/decode_st.yaml
+
+src_nbpe=1000
+tgt_nbpe=1000
+
+src_case=tc.rm
```
Comment (sw005320) [inline egs2/iwslt22_dialect/st1/run.sh:21]: It is better to add a comment of what is `tc.rm`, `tc`, and what options are available.

**Review by sw005320 — COMMENTED** [review]
LGTM
Please fix them according to my comments.

File: egs2/iwslt22_dialect/asr1/conf/tuning/train_asr_conformer_ctc0.3_lr2e-3_warmup15k_newspecaug.yaml (hunk 2 above)
Comment (sw005320) [reply egs2/iwslt22_dialect/asr1/conf/tuning/train_asr_conformer_ctc0.3_lr2e-3_warmup15k_newspecaug.yaml:1]: Oh, I see. You already put both numbers in the README file.
Sorry that I missed this.
Please ignore my comment then.

File: egs2/iwslt22_dialect/st1/RESULT.md (hunk 3 above)
Comment (sw005320) [reply egs2/iwslt22_dialect/st1/RESULT.md:9]: That's great!

Comment (brianyan918) [reply egs2/iwslt22_dialect/st1/RESULT.md]: this is done

File: egs2/iwslt22_dialect/st1/conf/decode_st_conformer.yaml [hunk 5]
```diff
@@ -0,0 +1 @@
+decode_pen/pen2.yaml
```
Comment (sw005320) [inline egs2/iwslt22_dialect/st1/conf/decode_st_conformer.yaml:1]: what's this?

File: egs2/iwslt22_dialect/st1/conf/decode_st_transformer.yaml [hunk 6]
```diff
@@ -0,0 +1 @@
+decode_pen/pen3.yaml
```
Comment (sw005320) [inline egs2/iwslt22_dialect/st1/conf/decode_st_transformer.yaml:1]: what's this?

File: egs2/iwslt22_dialect/st1/conf/decode_st_conformer.yaml (hunk 5 above)
Comment (brianyan918) [reply egs2/iwslt22_dialect/st1/conf/decode_st_conformer.yaml]: oops it was missing. thanks for catching it

**Comment by sw005320** [comment]
Thanks, @brianyan918!

**Pull request merged**